src/ocean/util/app/ext/TaskExt.d

```diff
@@ -0,0 +1,195 @@
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.util.app.ext.TaskExt;
+
+import ocean.transition;
+import ocean.util.app.ext.model.IConfigExtExtension;
+import ocean.task.Scheduler;
+import ocean.task.Task;
+import ocean.util.config.ConfigParser;
+import ocean.util.config.ConfigFiller;
+
+/// ditto
+class TaskExt : IConfigExtExtension
+{
+    /// Added to the configuration parsed from the config file if present
+    private SchedulerConfiguration.PoolDescription[] optional_config;
+
+    /// See `enabled`
+    private bool enabled_ = false;
+
+    /// Indicates that automatic scheduler configuration is enabled
+    public bool enabled ( )
+    {
+        return this.enabled_;
+    }
+
+    /***************************************************************************
+
+        Used to enable automatic handling of scheduler configuration and task
+        startup. Designed to be opt-in method to avoid any accidental breakage.
+
+        Params:
+            optional_config = if non-null, will be used as part of scheduler
+                configuration to init it
+
+    ***************************************************************************/
+
+    public void enable ( typeof(SchedulerConfiguration.specialized_pools)
+            optional_config = null )
+    {
+        this.optional_config = optional_config;
+        this.enabled_ = true;
+    }
+
+    /***************************************************************************
+
+        Parse the configuration file options to get the scheduler configuration
+
+        Params:
+            app = the application instance
+            parser = configuration instance
+
+    ***************************************************************************/
+
+    public override void processConfig ( IApplication app, ConfigParser parser )
+    {
+        if (!this.enabled)
+            return;
+
+        static struct PartialSchedulerConfiguration
+        {
+            size_t worker_fiber_stack_size =
+                SchedulerConfiguration.worker_fiber_stack_size.init;
+            size_t worker_fiber_limit =
+                SchedulerConfiguration.worker_fiber_limit.init;
+            size_t task_queue_limit =
+                SchedulerConfiguration.task_queue_limit.init;
+            size_t suspended_task_limit =
+                SchedulerConfiguration.suspended_task_limit.init;
+        }
+
+        // can't fill SchedulerConfiguration directly because it contains
+        // `PoolDescription` array which has to be provided via code
+
+        SchedulerConfiguration scheduler_config;
+        PartialSchedulerConfiguration partial_config;
+        // fill with init values
+        foreach (idx, ref elem; partial_config.tupleof)
+            elem = scheduler_config.tupleof[idx];
```

Comment:
Perhaps we should `assert` that the names of the fields are the same, otherwise bugs can cripple in if you change `SchedulerConfiguration` definition, but still have the matching field types.

Reply:
Even better - use reflection to access required fields by name ;)